azkaban-web-server/src/main/java/azkaban/webapp/servlet/AbstractAzkabanServlet.java

```diff
@@ -111,9 +92,7 @@ public AzkabanServer getApplication() {
 
   @Override
   public void init(final ServletConfig config) throws ServletException {
-    this.application =
-        (AzkabanServer) config.getServletContext().getAttribute(
-            AzkabanServletContextListener.AZKABAN_SERVLET_CONTEXT_KEY);
+    this.application = SERVICE_PROVIDER.getInstance(AzkabanWebServer.class);
```

Comment:
Will we want to inject AzkabanWebServer in the future? Do you expect any complications when if we do so?